### ISSUE TITLE
chainId address mismatch

### ISSUE THREAD

**Issue by M-Marcel** [PatrickAlphaC/ethers-simple-storage-fcc#20]
Hello guys, I got to the session were we are to deploy the SimpleStorage.sol contract with only transaction data, but I'm stuck with the following error.

Error: chainId address mismatch (argument="transaction", value={"nonce":3,"gasPrice":20000000000,"gasLimit":10000000,"to":null,"value":0,"data":"0x..."
reason: 'chainId address mismatch',
  code: 'INVALID_ARGUMENT',
  argument: 'transaction',

Here is the code snippet
```
const nonce = await wallet.getTransactionCount();
  const tx = {
    nonce: nonce,
    gasPrice: 20000000000,
    gasLimit: 10000000,
    to: null,
    value: 0,
    data: "0x...",
    chainId: 31337,
  };
  const sendTxResponse = await wallet.sendTransaction(tx);
  await sendTxResponse.wait(1);
  console.log(sendTxResponse);
  ```
  
  I tried using my ganache Network ID: 5777 as chainId as instructed but still got the same error.
  I also tried using 31337 as chainId but it didn't work.
  
  Any Ideas???

**Comment by deepakdubey8756** [comment]
Did you tried using 1337 as chain id.

**Comment by matischroder** [comment]
Usually is 1337. So as not to make a mistake, look directly at ganache gui.
![Screenshot from 2022-06-28 15-57-44](https://user-images.githubusercontent.com/37404887/176261291-d4f1f43b-5ad7-4126-86e6-67a0d33a6142.png)

**Comment by bensacc** [comment]
I had the same issue, and just commented out the chain id from the const tx assignment, which seems to have forced it to default to the correct one.  which is 1337, even though I have a network ID of 5777 in Ganache (both failed).  maybe just some kind of bug in Ganache?

**Comment by shawnesquivel** [comment]
TLDR; try 1337 as the chainId

**What I did**

I had the same issue. I used the Network Id of 5777 as the ChainId which was displayed in the Ganache UI.
![image](https://user-images.githubusercontent.com/94336773/206779914-1123f53a-2f52-4f0f-be57-8ede419c873c.png)

I got the same error as you, then realized that when I had deployed my ganache server (in a new terminal, 1) yarn add ganache
2) yarn ganache)
![image](https://user-images.githubusercontent.com/94336773/206780013-e5557ac6-3a03-4c37-a864-fc834a5e41ab.png)
This shows that the actual chainId is 1337. Alternatively, for some people 31337 will work but for you it didn't.
